### ISSUE TITLE
Darwin ARM64 Support

### ISSUE THREAD

**Issue by coolaj86** [goreleaser/goreleaser#1952]
I noticed that builds for Apple Silicon / M1 / ARM64 are not being built with goreleaser, even though I have go1.16 installed, which supports them.

```bash
go version
```

```
go1.16beta1 darwin/amd64
```

```bash
GOOS=darwin GOARCH=arm64 go build .
```

This is relevant to me in particular because I work on [webinstall.dev](https://webinstall.dev), which already supports M1 / Apple Silicon builds as soon as they are published, but many of the supported Go projects are using Goreleaser.

**Describe the solution you'd like**

I imagine that there's a simple line in a file somewhere to turn on darwin arm64 builds for Go versions that support it.

**Describe alternatives you've considered**

I can build by hand for the time being, or maybe fork goreleaser and hunt for the darwin options.

**Additional context**

N/A

**Comment by caarlos0** [comment]
Currently goreleaser allows only stable goos/goarches, but I guess it would be nice to have a beta feature for this... 

also need to handle brew arm64, for example.

**Comment by coolaj86** [comment]
Why not introspect what Go can build and use that as the list rather than hard-coding it?

**Comment by coolaj86** [comment]
Using `go tool dist list`:

```txt
aix/ppc64
android/386
android/amd64
android/arm
android/arm64
darwin/amd64
darwin/arm64
dragonfly/amd64
freebsd/386
freebsd/amd64
freebsd/arm
freebsd/arm64
illumos/amd64
ios/amd64
ios/arm64
js/wasm
linux/386
linux/amd64
linux/arm
linux/arm64
linux/mips
linux/mips64
linux/mips64le
linux/mipsle
linux/ppc64
linux/ppc64le
linux/riscv64
linux/s390x
netbsd/386
netbsd/amd64
netbsd/arm
netbsd/arm64
openbsd/386
openbsd/amd64
openbsd/arm
openbsd/arm64
openbsd/mips64
plan9/386
plan9/amd64
plan9/arm
solaris/amd64
windows/386
windows/amd64
windows/arm
```

**Comment by stack72** [comment]
I think limiting to Go 1.16.x is already causing a lot of us pain in supporting our users who are on the new apple macbooks...

**Comment by caarlos0** [comment]
Sorry @stack72 , didn't get what you meant... can you explain?

**Comment by stack72** [comment]
My apologies - it sent before my second sentence

I am using Go 1.15.3 locally and my "go tool diet list" supports Darwin/arm64 so it feels like waiting till 1.16 is released penalises the users that want to support arm64 now

**Comment by caarlos0** [comment]
On go <1.16 darwin/arm64 was ios, not macos. You can try to build anything locally and unless you have the right xcode stuff installed it wont work, if it does, it will not run on macs with apple silicon.

We're waiting on go 1.16 because it will be the first version to support macos on arm.

**Comment by caarlos0** [comment]
ex:

```sh
~/Developer/goreleaser/goreleaser brew-binary-tmpl ≡
❯ GOOS=darwin GOARCH=arm64 go build .
# github.com/goreleaser/goreleaser
/usr/local/Cellar/go/1.15.6/libexec/pkg/tool/darwin_amd64/link: running clang failed: exit status 1
ld: warning: ignoring file /var/folders/cz/k46fpqfd547_mn0ssjq7b5rr0000gn/T/go-link-407412967/go.o, building for macOS-x86_64 but attempting to link with file built for unknown-arm64
Undefined symbols for architecture x86_64:
  "_main", referenced from:
     implicit entry/start for main executable
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)


~/Developer/goreleaser/goreleaser brew-binary-tmpl ≡ 12s
❯ GOOS=darwin GOARCH=arm64 go1.16beta1 build .

~/Developer/goreleaser/goreleaser brew-binary-tmpl ≡ 22s
❯ file goreleaser
goreleaser: Mach-O 64-bit executable arm64
```

**Comment by stack72** [comment]
Thank you for the detailed information - this now helps clear up the issue for me!

**Comment by cjdenio** [comment]
Go 1.16 is stable! :tada: https://blog.golang.org/go1.16

**Comment by caarlos0** [comment]
```
❯ docker pull docker.io/library/golang:1.16-alpine
Error response from daemon: manifest for golang:1.16-alpine not found: manifest unknown: manifest unknown
```

doesn't exist yet :/

**Comment by caarlos0** [comment]
will use rc1 just so we release the stable binaries at least
